lighthouse-core/gather/gatherers/js-usage.js

```diff
@@ -34,9 +51,22 @@ class JsUsage extends Gatherer {
     /** @type {Record<string, Array<LH.Crdp.Profiler.ScriptCoverage>>} */
     const usageByUrl = {};
     for (const scriptUsage of scriptUsages) {
-      const scripts = usageByUrl[scriptUsage.url] || [];
+      // ScriptCoverages without a URL are from code we run over the protocol.
+      // We shouldn't analyze ourselves, so skip.
+      if (scriptUsage.url === '') continue;
+
+      // `ScriptCoverage.url` can be overridden by a magic sourceURL comment.
+      // Get the associated ScriptParsedEvent and use embedderName, which is the original url.
```

Comment:
can you describe embeddername a tad more and link up the CL that added it?

Reply:
Added a link but...what more is there to say?

Reply:
sure that's fine